src/scenario.jl

```diff
@@ -368,29 +366,37 @@ function run_model(domain::Domain, param_set::NamedDimsArray, corals::DataFrame,
 
     p.mb .= corals.mb_rate  # background mortality
     @set! p.k = max_cover  # max coral cover
-    @set! p.comp = sim_params.comp  # competition rate
 
     # Proportionally adjust initial cover (handles inappropriate initial conditions)
     proportional_adjustment!(Y_cover[1, :, :], cover_tmp, max_cover)
+    
+    # Define constant table location for seed values and seeded coral types
+    taxa_to_seed = [2,3,5]
+
+    taxa_names = param_set.factors[occursin.("N_seed_",param_set.factors)]
+
+    taxa_to_seed_ids = NamedDimsArray(corals.taxa_id .== taxa_to_seed',species=1:n_species,taxa=taxa_names)
 
-    # Define constant table location for seed values
-    tabular_enhanced::BitArray = corals.taxa_id .== 1
-    corymbose_enhanced::BitArray = corals.taxa_id .== 3
     target_class_id::BitArray = corals.class_id .== 2  # seed second smallest size class
-    seed_sc_TA::Int64 = first(findall(tabular_enhanced .& target_class_id))  # size class indices for TA and CA
-    seed_sc_CA::Int64 = first(findall(corymbose_enhanced .& target_class_id))
+    seed_sc = taxa_to_seed_ids .& target_class_id  # size class indices for TA, CA and SM
+
+    # Extract colony areas for sites selected in m^2 and add adaptation values
+    seeded_area = colony_mean_area(repeat(corals.mean_colony_diameter_m,1,3)[seed_sc]).*param_set(taxa_names)
```

Comment:
`seeded_area = colony_mean_area(repeat(corals.mean_colony_diameter_m,1,3)[seed_sc]).*param_set(taxa_names)`

This seems very convoluted. I'll simplify.